### DISCUSSION TITLE
Vue CLI Environment Variables

### DISCUSSION THREAD

**Discussion by LeoMartinDev** [vercel/vercel#4153] [General]
Vue CLI projects allows you to define `.env` files depending on your environment (during build, environment variables prefixed with `VUE_APP` will be replaced by the corresponding value).

Currently, I defined a `.env.production` in which I expose my production API URL (`VUE_APP_API_URL`).
It seems that Vercel does not care about my .env files and my variables are never defined.

See Vue CLI documentation about modes and environments : https://cli.vuejs.org/guide/mode-and-env.html#modes

**Comment by styfle** [comment]
Correct, Vercel does not upload `.env` or `.env.production` https://vercel.com/docs/v2/build-step#ignored-files-and-folders

We just announced the Environment Variables UI that you can use to configure production, preview, or development variables. https://vercel.com/blog/environment-variables-ui

You can use `now env pull` to create a `.env` file locally from your development environment variables defined in the cloud.
